linker-interface/shared/src/main/scala/org/scalajs/linker/interface/Linker.scala

```diff
@@ -23,8 +23,21 @@ import org.scalajs.logging.Logger
  *  .js file.
  */
 abstract class Linker private[interface] () {
-  def link(irFiles: Seq[IRFile],
+  final def link(irFiles: Seq[IRFile],
       moduleInitializers: Seq[ModuleInitializer],
       output: LinkerOutput, logger: Logger)(
+      implicit ec: ExecutionContext): Future[Unit] = {
+    val module = ModuleOutputSpec.Module("core")
+      .withInitializers(moduleInitializers: _*)
+
+    val spec = ModuleOutputSpec()
+      .addModule(ModuleOutputSpec.Selector.default, module)
+
+    val moduleOutput = ModuleLinkerOutput(module.id, output)
+
+    link(irFiles, spec, Seq(moduleOutput), logger)
+  }
+
+  def link(irFiles: Seq[IRFile], moduleSpec: ModuleOutputSpec, outputs: Seq[ModuleLinkerOutput], logger: Logger)(
```

Comment:
Note: The reason `moduleSpec` here is separate from `outputs` is that I anticipate most linker usages to be in a build system. In such a system, `moduleSpec` is provided by the user, but `outputs` is provided by the build system (probably dependent on `moduleSpec`).

This is less nice, because it will cause runtime failures, if the outputs are not in sync with what is in the `moduleSpec`.

Reply:
Right... There is actually the additional detail that a module does not need an internal module if no other modules depend on it. This constraint seems to not be easily mapped into the type system. So we need to validate outputs anyways.

That being said, the more type safety the better is of course also an argument :S

Reply:
It may not need a module-name, but it still needs an internal module, just so you know what classes to put in what file.